### ISSUE TITLE
sr-only-*-up / sr-only-*-down utility classes for accessibility

### ISSUE THREAD

**Issue by hjnilsson** [twbs/bootstrap#19954]
The most common use case for me when using the `hidden-*-down` and `-up` classes is to hide button labels on phone screens and similar use-cases. This is a sensible thing to do, and would result in markup like follows:

```
<a class="btn" href="https://google.com">
   <i aria-hidden="true" class="fa fa-search"></i>
   <span class="hidden-sm-down">Go to google</span>
</a>
```

The issue here, is that for accessibility features on the phone (like VoiceOver), the text of the button will now be hidden, and it will simply read "Button". This is fixed right now by using the `aria-label` tag on the button element:

```
<a class="btn" href="https://google.com" aria-label="Go to google">
   <i aria-hidden="true" class="fa fa-search"></i>
   <span class="hidden-sm-down">Go to google</span>
</a>
```

This will be read correctly even on mobile screens. However, this result in needless repetition in the HTML, when the same effect could be acheived if we had a `sr-only-sm-down` utility class, like follows: 

```
<a class="btn" href="https://google.com">
   <i aria-hidden="true" class="fa fa-search"></i>
   <span class="sr-only-sm-down">Go to google</span>
</a>
```

This will correctly read in VoiceOver on the iPhone, remaining universally accessible. I suggest these classes should be added to standard bootstrap 4. To make it easier to create accessible websites

**Comment by mejine** [comment]
The most common use case for me when using the ,!

**Comment by cvrebert** [comment]
CC: @patrickhlauke

**Comment by joe-watkins** [comment]
@hjnilsson I like this idea,.. but if it doesn't make it to core you could roll your own: [CodePen](http://codepen.io/joe-watkins/pen/MegwvP/)

**Comment by patrickhlauke** [comment]
seems sensible to me

**Comment by hjnilsson** [comment]
After some deliberation, I believe only `sr-only-*-down` is necessary. I have never had need for the reverse (ie. hide elements from display but not for screen readers on desktop). The use case for `hidden-xs-up` is where a more complex element would be shown for desktop, and a simpler version for mobile phones.

**Comment by Shnoulle** [comment]
Hi,

In fact : i think .hidden-*-down and .hidden-*-up must NEVER be hidden to screen reader. Maybe restrict to @media screen, or use another css.

And the !important is .... difficult.

**Comment by mdo** [comment]
We don't do the hidden up/down classes anymore. Would `.sr-only-{sm|md|lg|xl}` work still?

**Comment by patrickhlauke** [comment]
Late reply @mdo - what would the effect of, say, `.sr-only-md` be? for viewport sizes smaller than and up to  (but not including) `lg` have the thing act as `.sr-only`, and noop on `lg` and `xl` ? If so, then yes, this would still be useful.

**Comment by soniktrooth** [comment]
++ this would be super handy

**Comment by mdo** [comment]
Closing as stale.
